CI: teensy41 build fails to link HydroPack example (pre-existing on master since #3683)

## What

The `teensy41` CI workflow fails on current `master` (confirmed at commit 214d499e5f, the #3683 HydroPack merge) with a link error:

```
build error: build failed: arm-none-eabi-gcc link failed:
collect2: error: ld returned 1 exit status
❌ Compilation failed (fbuild) [3.4s]
FAILED: hydropack
ERROR: Compilation failed for board teensy41
```

Run: https://github.com/FastLED/FastLED/actions/runs/29476427917 (master, headSha 214d499e5f8bc8655013b3b48b5f72171dc02506)

## How discovered

Found while landing PR #3684 (fix for #3682) — the same `teensy41` failure reproduced identically on the PR branch (forked from the same master commit), confirming it's pre-existing and unrelated to that change: https://github.com/FastLED/FastLED/actions/runs/29477834805

## Suggested fix

Investigate the HydroPack example (`examples/HydroPack/`, added in #3683) for a teensy41-specific link failure — likely an unresolved symbol or missing platform guard for a peripheral/library not available on Teensy 4.1 (PWM/audio init added in that PR).

## Handoff — session ending (token budget)

This issue was filed as a byproduct of landing #3684 (fix for #3682); it was never investigated beyond confirming it's pre-existing on `master` and unrelated to that change. Picking this up fresh, here's what's known and what isn't.

### Confirmed facts
- Failure is 100% reproducible on `master` at commit `214d499e5f` (the #3683 HydroPack merge) — same link error on a totally unrelated PR branch forked from that same commit.
- Failure mode: `teensy41` build, linking the `hydropack` example fails:
  ```
  build error: build failed: arm-none-eabi-gcc link failed:
  collect2: error: ld returned 1 exit status
  FAILED: hydropack
  ERROR: Compilation failed for board teensy41
  ```
- Reference run: https://github.com/FastLED/FastLED/actions/runs/29476427917
- Not investigated: the actual undefined-symbol/missing-library detail. The log excerpt above is from `--log-failed` output which only showed the high-level fbuild summary, not the raw `arm-none-eabi-gcc`/`collect2` linker output (unresolved symbol names). That's the first thing the next session should pull — re-run `gh run view 29476427917 --log` (or a fresh CI run) and grep for `undefined reference` near the `collect2` failure to get the actual missing symbol(s).

### Not yet done / next steps
1. Get the real linker error (undefined symbols) — see above.
2. Read `examples/HydroPack/` (added in #3683) for anything platform-gated incorrectly — likely a peripheral/library (audio, PWM) that's available on ESP32/WASM but not guarded out on Teensy 4.1, or a Teensy-specific symbol that's declared but not implemented/linked.
3. Check whether `examples/HydroPack` should be excluded from the teensy41 compile matrix (if HydroPack is inherently ESP32/WASM-only) vs. actually fixed to build on Teensy41 — that's a scope decision, not a given.
4. Validate fix via `bash compile teensy41 --examples HydroPack` locally before pushing.

No worktree or branch was created for this issue — clean slate to start from.

--- a/examples/hydropack/hydropack.ino
+++ b/examples/hydropack/hydropack.ino
@@ -5,7 +5,7 @@
 ///          INMP441 mic -> normalized bass detection -> 50 Hz PWM gates.
 /// @example hydropack.ino
 
-// @filter: (mem is large)
+// @filter: (mem is large) and (platform is esp32*)
 
 // This sketch targets the ESP32-C3 and is also a valid FastLED WASM sketch:
 //   pip install fastled

